Mjml.Net/Components/Body/MsoButtonComponent.cs

```diff
@@ -0,0 +1,153 @@
+        };
+        public override string ComponentName => "mj-msobutton";
+
+        [Bind("mso-proof", BindType.Boolean)]
+        public string? MsoProof = "false";
+
+        [Bind("mso-height", BindType.PixelsOrPercent)]
+        public string? MsoHeight;
+
+        [Bind("mso-width", BindType.PixelsOrPercent)]
+        public string? MsoWidth;
+
+        public override void Render(IHtmlRenderer renderer, GlobalContext context)
+        {
+            var isMsoProof = MsoProof == "true";
+            if (isMsoProof)
+            {
+                RenderMso(renderer);
+                renderer.Content("<!--[if !mso]><!-->");
+                {
+                    base.Render(renderer, context);
+                }
+                renderer.Content("<!--<![endif]-->");
+            }
+            else
+            {
+                base.Render(renderer, context);
+            }
+        }
+
+        private void RenderMso(IHtmlRenderer renderer)
+        {
+            var borderAttributes = new[] { "0pt", "Solid", "#000000" };
+            var stroked = Border != "none";
+            var hasBackgroundColor = !string.IsNullOrEmpty(BackgroundColor) && BackgroundColor != "none";
+            if (stroked)
+            {
+                var border = Border.Split(" ");
+                borderAttributes = new[]
+                {
+                    border.Length > 0 ? border[0] : borderAttributes[0],
+                    border.Length > 1 ? BorderstyleAdapter[border[1]] : borderAttributes[1],
+                    border.Length == 3 ? border[2] : borderAttributes[2],
+                };
+            }
+
+            renderer.Content("<!--[if mso]>");
+            {
+                renderer.StartElement("tr");
+                renderer.StartElement("td")
+                    .Attr("align", Align)
+                    .Attr("bgcolor", BackgroundColor)
+                    .Attr("role", "presentation")
+                    .Attr("valign", VerticalAlign)
+                    .Style("border", Border)
+                    .Style("border-bottom", BorderBottom)
+                    .Style("border-left", BorderLeft)
+                    .Style("border-right", BorderRight)
+                    .Style("border-top", BorderTop)
+                    .Style("border-radius", BorderRadius)
+                    .Style("cursor", "auto")
+                    .Style("font-style", FontStyle)
+                    .Style("height", Height)
+                    .Style("mso-padding-alt", InnerPadding)
+                    .Style("text-align", TextAlign)
+                    .Style("background", BackgroundColor);
+
+                renderer.StartElement("v:roundrect")
+                    .Attr("xmlns:v", "urn:schemas-microsoft-com:vml")
+                    .Attr("xmlns:w", "urn:schemas-microsoft-com:office:word")
+                    .Attr("fill", hasBackgroundColor ? "t" : "f")
+                    .Attr("strokeweight", stroked ? borderAttributes[0] : "0pt")
```

Comment:
is this "if" here really needed, you already have it above.

if "stroked" is not true, the width is 0pt anyway.

Reply:
yes, you right, "if" removed